mono/mini/aot-runtime.h

```diff
@@ -244,6 +244,9 @@ void     mono_aot_set_make_unreadable       (gboolean unreadable);
 gboolean mono_aot_is_pagefault              (void *ptr);
 void     mono_aot_handle_pagefault          (void *ptr);
 void     mono_aot_register_jit_icall        (const char *name, gpointer addr);
+#ifdef __cplusplus
+#define mono_aot_register_jit_icall(name, addr) (mono_aot_register_jit_icall ((name), (gpointer)(addr)))
+#endif
```

Comment:
I think we should fix all the callsites instead. the aren't that many.

Reply:
Ok, but it is a somewhat recurring thing, i.e. in larger numbers, in PR not submitted yet.

Reply:
Ok I did that. There were around 19. How about mono_emit_jit_icall? Around 80 of those.